modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaUpdateStatusMethodCheck.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaUpdateStatusMethodCheck
+	extends BaseUpgradeMatcherReplacementCheck {
+
+	@Override
+	protected String formatMatcherIteration(
+		String content, String newContent, Matcher matcher) {
+
+		String methodCall = JavaSourceUtil.getMethodCall(
+			content, matcher.start());
+
+		String variable = matcher.group(2);
+
+		if (!methodCall.contains("UserLocalServiceUtil") &&
+			!hasClassOrVariableName(
+				"UserLocalService", content, content, variable) &&
+			!hasClassOrVariableName(
+				"UserService", content, content, variable)) {
+
+			return newContent;
+		}
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
+
+		if (parameterList.size() != 2) {
+			return content;
+		}
+
+		if (!hasParameterTypes(
+				content, content, ArrayUtil.toStringArray(parameterList),
+				new String[] {"long", "int"})) {
+
+			addMessage(
+				content,
+				StringBundler.concat(
+					"Could not resolve types of updateStatus method. The ",
+					"method signature has changed to updateStatus(long userId,",
+					"int status, ServiceContext serviceContext). Fill the new ",
+					"parameter manually."));
+		}
+
+		StringBundler sb = new StringBundler(4);
+
+		sb.append(JavaSourceUtil.getIndent(methodCall));
+		sb.append("ServiceContext serviceContextThreadLocal = ");
+		sb.append("ServiceContextThreadLocal.getServiceContext();\n\n");
+		sb.append(matcher.group(1));
+
+		String newMethod = JavaSourceUtil.addMethodNewParameters(
+			parameterList.size(), methodCall, variable,
+			Arrays.asList("(ServiceContext)serviceContextThreadLocal.clone()"),
+			sb.toString());
+
+		return StringUtil.replace(newContent, methodCall, newMethod);
+	}
+
+	@Override
+	protected Pattern getPattern() {
+		return Pattern.compile(
+			"(|\\t*\\w+\\s*\\=|\\t*return?)(\\t*\\s?\\w+\\.updateStatus\\()");
+	}
```

Comment:
Hi @TamyrisTorres, the `?` is unnecessary in this regex, no? 
The first one is one make the "n" of the "return" optional is this correct? The second \\s? is about only one space. Why you don't use `*` instead?

Reply:
In this case, I use '?' because with '*' the regex gets the line above the method but I don´t want. I need just the '/t' on the line. 

![Selection_017](https://github.com/nicolas-sss/liferay-portal/assets/89744445/20359c47-04f4-474d-8390-e4155de53930)

![Selection_018](https://github.com/nicolas-sss/liferay-portal/assets/89744445/ab9b5b8b-093c-4a28-adfc-6fad6437e342)